CMakeLists.txt

```diff
@@ -1,29 +1,38 @@
-cmake_minimum_required(VERSION 2.8.12)
+#TODO
+# - fix / modernize cmake
+# - lower cmake version a bit ~3.7 to support older systems
+# - fix abseil warnings or try to use SYSTEM / -isystem includes
+
+cmake_minimum_required(VERSION 3.8)
 project(s2-geometry)
+
+#produces compile_command.json
 set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
+
+#not used?
+#list(APPEND CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/third_party/cmake")
+#include(CheckCXXCompilerFlag)
+#include(FindPackageHandleStandardArgs)
+
+## set options
 include(CMakeDependentOption)
-include(CheckCXXCompilerFlag)
```

Comment:
Separate PR to remove unneeded options.

Reply:
I am not sure if we will get on the same page here. I favor to rewrite the whole file on my branch and squash everything. I am willing to reason with you about every change, pass all tests and try to find the best solution. I do not want to create a meaningful commit with message for every line. That would be a too big time investment for me. Of course I will remove unrelated changes like in `coder.i`. And eventually smaller commits could be created when once the work is done.